### PR TITLE
Converts codemirror to service providers

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Converts the codemirror editor to service providers.

This pr contains the commits from #39625.

### Testing Instructions
- Make sure Codemirror is the default editor
- Edit an article

Also test if the other editors do still work

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: plugins/editors/codemirror/src/Extension/Codemirror.php [hunk 1]
```diff
@@ -60,15 +61,7 @@ class PlgEditorCodemirror extends CMSPlugin
...
-     * Application object.
-     *
-     * @var    \Joomla\CMS\Application\CMSApplication
-     * @since  4.0.0
-     */
-    protected $app;
-
-    /**
-     * Initialises the Editor.
+     * Initializes the Editor.
```
Comment (brianteeman) [inline plugins/editors/codemirror/src/Extension/Codemirror.php]:
```suggestion
     * Initialises the Editor.
```

**Comment by dgrammatiko** [comment]
I have tested this item :white_check_mark: successfully on 36f9ebdb0949d6e2d776bcaf1fd6f4c99037d802<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39638">issues.joomla.org/tracker/joomla-cms/39638</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 36f9ebdb0949d6e2d776bcaf1fd6f4c99037d802<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39638">issues.joomla.org/tracker/joomla-cms/39638</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39638">issues.joomla.org/tracker/joomla-cms/39638</a>.</sub>

**Comment by obuisard** [comment]
Thank you Allon @laoneo !

**Pull request merged**